components/automate-cli/pkg/verifyserver/services/softwareversionservice/softwareversionservice.go

```diff
@@ -0,0 +1,195 @@
+        cmdCheckArray: cmdCheckArray,
+        osFilepath:    osFilepath,
+    }
+}
+
+const osFilepath = "/etc/os-release"
+
+var cmdCheckArray = []string{"mkdir", "useradd", "chown", "rm", "touch", "truncate", "echo", "sleep", "ls", "grep", "yum", "which", "cp", "curl", "bash", "sysctl", "cat", "sed", "mount", "pvcreate", "vgcreate", "lvcreate", "mv", "systemd", "wget", "exec"}
+
+func (sv *SoftwareVersionService) GetSoftwareVersionServices(query string) (models.SoftwareVersionDetails, error) {
+    serviceResponse := models.SoftwareVersionDetails{}
+    serviceResponse.Passed = true
+    serviceResponseArray := []models.Checks{}
+    cmdArray := []string{}
+    if query == "postgres" {
+        cmdArray = []string{"stat"}
+        cmdArray = append(cmdArray, sv.cmdCheckArray...)
+    } else if query == "opensearch" {
+        cmdArray = []string{"openssl"}
+        cmdArray = append(cmdArray, sv.cmdCheckArray...)
+    } else {
+        cmdArray = append(cmdArray, sv.cmdCheckArray...)
+    }
+
+    for i := 0; i < len(cmdArray); i++ {
+        checkResponse := checkCommandVersion(cmdArray[i])
+        if !checkResponse.Passed {
+            serviceResponse.Passed = false
+        }
+        serviceResponseArray = append(serviceResponseArray, checkResponse)
+    }
+    osResponse, err := checkOsVersion(sv.osFilepath)
+    if err != nil {
+        logger.NewLogrusStandardLogger().Error("Error while getting OS version = ", err)
+        return models.SoftwareVersionDetails{
+            Passed: false,
+            Checks: []models.Checks{
+                {},
+            },
+        }, err
+    }
+    if !osResponse.Passed {
+        serviceResponse.Passed = false
+    }
+    serviceResponseArray = append(serviceResponseArray, osResponse)
+    serviceResponse.Checks = serviceResponseArray
+    return models.SoftwareVersionDetails{
+        Passed: serviceResponse.Passed,
+        Checks: serviceResponse.Checks,
+    }, nil
+}
+func checkCommandVersion(cmdName string) models.Checks {
+    _, err := exec.LookPath(cmdName)
+    if err != nil {
+        return models.Checks{
+            Title:         cmdName + " availability",
+            Passed:        false,
+            SuccessMsg:    "",
+            ErrorMsg:      cmdName + " is not available",
+            ResolutionMsg: "Ensure " + cmdName + " is available in $PATH on the node",
+        }
+    }
+    return models.Checks{
+        Title:         cmdName + " availability",
+        Passed:        true,
+        SuccessMsg:    cmdName + " is available",
+        ErrorMsg:      "",
+        ResolutionMsg: "",
+    }
+}
+
+func readFile(osFilepath string) ([]byte, error) {
+    data, err := ioutil.ReadFile(osFilepath) // nosemgrep
+    if err != nil {
+        logger.NewLogrusStandardLogger().Error("Error while reading the OS file from the path = ", err)
+        return []byte{}, err
+    }
+    return data, nil
+}
+
+func getOsVersion(osFilepath string) (string, string, error) {
```

Comment:
move these functions to our utils

Reply:
Done moved to getosutis inside utis folder